src/IO/ITKImageInputFileFormat.cxx

```diff
@@ -140,76 +120,261 @@ read_from_file(const std::string& filename) const
     (read_file_itk< STIRImageType >(filename));
 }
 
-// Actual conversion function
-STIRImageSingle*
-convert_ITK_to_STIR(const ITKImageSingle::Pointer itk_image_orig)
+/* Convert ITK (LPS) coordinates into STIR physical coordinates and
+   accounting for the change of origin by default.
+
+   However, if `is_displacement_field` is true, coordinates are
+   interpreted as being displacement vectors and hence the change of
+   origin is ignored.
+ */
+template<typename ITKImageType, typename STIRImageType>
+CartesianCoordinate3D<float>
+ITK_coordinates_to_STIR_physical_coordinates
+(const typename ITKImageType::PointType &itk_coord,
+ const STIRImageType &stir_image, bool is_displacement_field = false)
 {
-    ITKImageSingle::Pointer itk_image;
-    CartesianCoordinate3D<float> voxel_size;
-    BasicCoordinate<3,int> min_indices, max_indices;
-    CartesianCoordinate3D<float> origin;
-
-    // orientate the ITK image
-    orient_ITK_image<ITKImageSingle>(
-                itk_image, voxel_size, min_indices, max_indices, origin, itk_image_orig);
-
-    // create STIR image
-    VoxelsOnCartesianGrid<float>* image_ptr =
-        new VoxelsOnCartesianGrid<float>
-            (IndexRange<3>(min_indices, max_indices),
-            origin,
-            voxel_size);
-
-    // copy data
-    VoxelsOnCartesianGrid<float>::full_iterator stir_iter = image_ptr->begin_all();
-    typedef itk::ImageRegionConstIterator< ITKImageSingle > IteratorType;
-    IteratorType it (itk_image, itk_image->GetLargestPossibleRegion() );
-    for ( it.GoToBegin(); !it.IsAtEnd(); ++it, ++stir_iter  ) {
-        *stir_iter = static_cast<float>(it.Get());
-    }
+  // find STIR origin
+  // Note: need to use - for z-coordinate because of different axis conventions
+  CartesianCoordinate3D<float> stir_coord
+    = stir_image.get_physical_coordinates_for_LPS_coordinates
+      (CartesianCoordinate3D<float>(static_cast<float>(itk_coord[2]),
+                                    static_cast<float>(itk_coord[1]),
+                                    static_cast<float>(itk_coord[0])));
+
+  // The following is not required for displacement field images, as
+  // the coordinates are relative.
+  if (!is_displacement_field)
+  {
+    CartesianCoordinate3D<float> stir_origin_index(0, 0, 0);
```

Comment:
I don't understand this logic. Shouldn't the origins of all images (3D, displacement or deformation) should all be altered?

Reply:
Ah, sure, sorry.